phpBB/phpbb/mention/helper/mention_helper.php

```diff
@@ -0,0 +1,196 @@
+                $new_post_text = $new_post_text . $strip_mention;
+                $new_post_text = $new_post_text . $add_url_tag;
+                $prev_end_pos = $endpos + 1;
+            }
+            //maintain al list of all different userids.
+            if (!in_array($userid_list[$username_clean], $users_already_mapped, true))
+            {
+                array_push($users_already_mapped, $userid_list[$username_clean]);
+            }
+        }
+        return array("post_text" => $new_post_text, "users_mapped" => $users_already_mapped) ;
+    }
+
+    /**
+    * Function to extract the mentioned users and replace them with links to the user
+    * profile
+    *
+    * @param string   $post              post text
+    * @param array    $data              Notification details
+    * @param object   $notif_manager_obj  Notification Manager object
+    *
+    * @return array Array of responder data
+    */
+    public function get_mentioned_users($post_text, $data, $notif_manager_obj)
+    {
+        $this->data = $data;
+        $regular_expression_match = '#\[mention\](.*?)\[/mention\]#';
+        $matches = false;
+        $matches = $this->get_regex_match($regular_expression_match, $post_text);
+        if (count($matches[1]) > 0)
+        {
+            $start_tag_length = strlen("[mention]");
+            $end_tag_length = strlen("[\mention]");
+            $user_list = array();
+            $userid_list = array();
+            $user_list = $this->get_user_list($matches);
+            if (count($user_list) > 0)
+            {
+                $temp_notif_type_object = $notif_manager_obj->get_item_type_class("notification.type.mention");
+                $userid_list = $temp_notif_type_object->find_users_for_notification($this->db, $user_list);
+                if (count($userid_list) > 0)
+                {
+                    $new_post_data = $this->get_regex_substituted_text($matches, $post_text, $start_tag_length, $end_tag_length, $userid_list);
+                    return array("new_post_text" => $new_post_data["post_text"], "users_mentioned" => $new_post_data["users_mapped"], "notif_type_object" => $temp_notif_type_object);
+                }
+            }
+        }
+        return $post_text;
+    }
+
+    /**
+    * Function to generate Notifications.
+    *
+    * @param $user_list    array           Array containing userids to send
+    *                                       notifications to.
+    * @param $notif_manager_obj  object   object    Notification Manager object
+    * @param $temp_notif_type_object      object    Temporary notiifcation type object\
+    *                                                to call the method from
+    *                                                new notiifcation type - mention.
+    *
+    */
+    public function send_notifications($user_list, $notif_manager_obj, $temp_notif_type_object)
```

Comment:
Also, this function seems to duplicates the notification manager logic. You should uses it instead. Or it isn't possible? Explain me in that case